functions/java/modules/module1/src/main/java/com/amazonaws/powertools/workshop/Utils.java

```diff
@@ -0,0 +1,183 @@
+        LOGGER.info("Task in status {}", status);
+        String graphQLMutationResponse = performGraphQLMutation(APPSYNC_ENDPOINT, UPDATE_FILE_STATUS_QUERY.formatted(UPDATE_FILE_STATUS_MUTATION, fileId, status,
+                transformedFileKey != null ? ", \"transformedFileKey\": \"" + transformedFileKey + "\"" : ""));
+        LOGGER.debug(graphQLMutationResponse);
+    }
+
+    private static String performGraphQLMutation(String endpoint, String jsonQuery) {
+        LOGGER.debug(jsonQuery);
+
+        String host;
+        String path;
+        try {
+            host = new URL(endpoint).getHost();
+            path = new URL(endpoint).getPath();
+        } catch (MalformedURLException e) {
+            throw new UpdateFileStatusException(e);
+        }
+
+        SdkHttpFullRequest request = SdkHttpFullRequest
+                .builder()
+                .method(SdkHttpMethod.POST)
+                .putHeader("Host", host)
+                .putHeader("Content-Type", "application/json")
+                .host(host)
+                .encodedPath(path)
+                .protocol("HTTPS")
+                .contentStreamProvider(() -> new ByteArrayInputStream(jsonQuery.getBytes(StandardCharsets.UTF_8)))
+                .build();
+
+        ExecutionAttributes attributes = ExecutionAttributes.builder()
+                .put(AwsSignerExecutionAttribute.AWS_CREDENTIALS, credentialsProvider.resolveCredentials())
+                .put(AwsSignerExecutionAttribute.SERVICE_SIGNING_NAME, "appsync")
+                .put(AwsSignerExecutionAttribute.SIGNING_REGION, AWS_REGION)
+                .build();
+
+        // Perform AWS Signature Version 4 signing
+        Aws4Signer signer = Aws4Signer.create();
+        SdkHttpFullRequest prepRequest = signer.sign(request, attributes);
+
+        // Perform the HTTP call to GraphQL endpoint
+        HttpPost httpPost = new HttpPost(endpoint);
+        for (Map.Entry<String, List<String>> header : prepRequest.headers().entrySet()) {
+            if (header.getKey().equalsIgnoreCase("host")) { continue; }
+            for (var value : header.getValue()) {
+                httpPost.addHeader(header.getKey(), value);
+            }
+        }
+
+        try (CloseableHttpClient client = HttpClients.custom().build()) {
+            httpPost.setEntity(new ByteArrayEntity(jsonQuery.getBytes(StandardCharsets.UTF_8)));
+            return performHttpCall(client, httpPost);
+        } catch (IOException e) {
+            throw new UpdateFileStatusException(e);
+        }
+    }
+
+    private static String performHttpCall(CloseableHttpClient client, HttpPost httpPost) {
+        try (var response = client.execute(httpPost)) {
+            String responseData = new String(response.getEntity().getContent().readAllBytes(), StandardCharsets.UTF_8);
+            if (responseData.contains("errors")) {
+                throw new UpdateFileStatusException(responseData);
```

Comment:
The generic name of the method `performHttpCall` janks with it being tightly coupled to updating file statuses